test/gitea_test.go

```diff
@@ -894,6 +897,50 @@ func TestGiteaErrorSnippet(t *testing.T) {
 	tgitea.WaitForPullRequestCommentMatch(t, topts)
 }
 
+func TestGiteaErrorSnippetCustomLines(t *testing.T) {
+	ctx, _ := rtesting.SetupFakeContext(t)
+	topts := &tgitea.TestOpts{
+		TargetEvent: triggertype.PullRequest.String(),
+		YAMLFiles: map[string]string{
+			".tekton/pr.yaml": "testdata/pipelinerun-error-snippet.yaml",
+		},
+		CheckForStatus:   "failure",
+		ExpectEvents:     false,
+		SkipEventsCheck:  true,
+		StatusOnlyLatest: true,
+	}
+	topts.TargetRefName = names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("pac-e2e-test")
+	topts.TargetNS = topts.TargetRefName
+	topts.ParamsRun, topts.Opts, topts.GiteaCNX, _ = tgitea.Setup(ctx)
+	assert.NilError(t, topts.ParamsRun.Clients.NewClients(ctx, &topts.ParamsRun.Info))
+	ctx, err := cctx.GetControllerCtxInfo(ctx, topts.ParamsRun)
+	assert.NilError(t, err)
+	assert.NilError(t, pacrepo.CreateNS(ctx, topts.TargetNS, topts.ParamsRun))
+	cfgMapData := map[string]string{
+		"error-log-snippet-number-of-lines": "5",
+	}
+	defer configmap.ChangeGlobalConfig(ctx, t, topts.ParamsRun, cfgMapData)()
+
+	_, f := tgitea.TestPR(t, topts)
+	defer f()
+
+	topts.Regexp = regexp.MustCompile(`Hey man i just wanna to say i am not such a failure, i am useful in my failure`)
```

Comment:
I said the same thing a lot in college to myself in the mirror, during exam season /s